mbed_flasher/flashers/FlasherJLink.py

```diff
@@ -0,0 +1,142 @@
+"""
+Copyright 2016 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from distutils import spawn
+try:
+    import queue
+except ImportError:
+    import Queue as queue
+import sys
+import tempfile
+
+import mbed_lstools
+
+from mbed_flasher.flashers.FlasherBase import FlasherBase
+from mbed_flasher.return_codes import EXIT_CODE_SUCCESS
+from mbed_flasher.return_codes import EXIT_CODE_FLASH_FAILED
+
+
+class FlasherJLink(FlasherBase):
+    """
+    Class FlasherJLink
+
+    Target on flashing JLink boards with JLinkExe.
+    """
+    name = "jlink"
+    executable = "JLink.exe" if sys.platform.startswith("win") else "JLinkExe"
+    supported_targets = None
+
+    def __init__(self, logger=None):
+        FlasherBase.__init__(self, logger)
+
+    @staticmethod
+    def get_supported_targets():
+        """
+        :return: supported JLink types
+        """
+        if not FlasherJLink.supported_targets:
+            mbeds = mbed_lstools.create()
+            # @todo this is workaround until mbed-ls provide public api
```

Comment:
is there task for mbedls for this ?

Reply:
Did you make one when you changed it?

Reply:
ah this is same one? then there should be - or is there.. hmm. Need to check and raise if not exists..